pkg/server/options/options.go

```diff
@@ -204,20 +204,36 @@ func (o *Options) Complete() (*CompletedOptions, error) {
 		return nil, err
 	}
 
+	var err error
 	if !filepath.IsAbs(o.EmbeddedEtcd.Directory) {
-		o.EmbeddedEtcd.Directory = filepath.Join(o.Extra.RootDirectory, o.EmbeddedEtcd.Directory)
+		o.EmbeddedEtcd.Directory, err = filepath.Abs(o.EmbeddedEtcd.Directory)
+		if err != nil {
+			return nil, err
+		}
 	}
 	if !filepath.IsAbs(o.GenericControlPlane.SecureServing.ServerCert.CertDirectory) {
-		o.GenericControlPlane.SecureServing.ServerCert.CertDirectory = filepath.Join(o.Extra.RootDirectory, o.GenericControlPlane.SecureServing.ServerCert.CertDirectory)
+		o.GenericControlPlane.SecureServing.ServerCert.CertDirectory, err = filepath.Abs(o.GenericControlPlane.SecureServing.ServerCert.CertDirectory)
+		if err != nil {
+			return nil, err
+		}
 	}
 	if len(o.GenericControlPlane.SecureServing.ServerCert.CertDirectory) == 0 {
-		o.GenericControlPlane.SecureServing.ServerCert.CertDirectory = filepath.Join(o.Extra.RootDirectory, "certs")
+		o.GenericControlPlane.SecureServing.ServerCert.CertDirectory, err = filepath.Abs("certs")
```

Comment:
it doesn't (copy&paste). And it was wrong before too I think. Fixing.

Reply:
please take another look. The old approach didn't work. If you have better ideas, am welcome.

Reply:
"If the cert directory is not absolute, override to to "certs" and make it absolute"?

Reply:
no, moved it into the normal place for defaulting.

Reply:
ok, old code created directly in root-dir because isAbs above was false for empty string, and then concatenation with rootDir happened. This code here was dead. Removed it now and made the behaviour explicit. Let's see whether e2e are fine (VW tests grab the apiserver.crt+key from the rootDir).

Reply:
It still looks like it's missing defaulting to "certs"?

Reply:
see the NewOptions func

Reply:
but as written above, the old behaviour was not "certs", but rootDir directly. So that's what I added as default now.